src/battle_anim_new.c

```diff
@@ -9154,16 +9154,15 @@ void AnimTask_SyrupBomb(u8 taskId)
     struct Pokemon *party = GetBattlerParty(gBattleAnimAttacker);
     u32 isShiny = IsMonShiny(&party[gBattlerPartyIndexes[gBattleAnimAttacker]]);
 
-    gDisableStructs[gBattleAnimTarget].syrupBombIsShiny = isShiny;
+    if (!(gStatuses4[gBattleAnimTarget] & STATUS4_SYRUP_BOMB))
+        gDisableStructs[gBattleAnimTarget].syrupBombIsShiny = isShiny;
```

Comment:
Again, you need to test the specific case where the target is first hit by a non-shiny Syrup Bomb and then a shiny one in a link battle. `gStatuses4` is never updated on the slave GBA, so the second time the target gets hit by the attack, on one GBA the shiny animation will start to play (since the condition `!(gStatuses4[gBattleAnimTarget] & STATUS4_SYRUP_BOMB)` will always be true).

Reply:
Yeah, you are correct! Sorry, I didn't fully understand what you meant previously.

 What would be a good replacement in this case?

Reply:
I pushed a commit with current changes but I can't get it to work
![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/0b5d3d98-0fdf-4059-8ebb-fec299f5c453)

Reply:
Okay, so this still is not working because of how the special battle animations are set up. Unlike when a move animation is used, the disable struct is _not_ passed along to the special battle animations.

To fix this, copy the line `memcpy(&sBattleBuffersTransferData[16], disableStructPtr, sizeof(struct DisableStruct));` from `BtlController_EmitMoveAnimation`, and add it to `BtlController_EmitBattleAnimation`, changing the `16` to wherever the current data struct leaves of (and modify the amount of data transferred over). Then, modify all `...HandleBattleAnimation` functions to accept the disable struct as well (see how the `...HandleMoveAnimation` functions do it). Finally, in the sticky syrup animation, you should be able to access the value via `gAnimDisableStructPtr->syrupBombIsShiny`.

Reply:
That worked, thank you!
![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/67ecaa18-031c-47e6-8211-fb6429536896)
Though I didn't know what I was doing and relied on your description only so not sure if all of it is correct. 😅